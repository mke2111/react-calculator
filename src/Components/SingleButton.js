/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React from 'react';
import '../index.css';

function SingleButton(props) {
  return (
    <button className="square" type="button">
      {props.name}
    </button>
  );
}

SingleButton.propType = {
  name: 0,
};

export default SingleButton;
