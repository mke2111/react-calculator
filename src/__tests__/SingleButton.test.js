import React from 'react';
import renderer from 'react-test-renderer';
import SingleButton from '../Components/SingleButton';

describe('Test Button component with snapshot', () => {
  it('Renders button component correctly', () => {
    const buttonComponent = renderer.create(<SingleButton btnName="+" clickHandler={() => '6'} />).toJSON();

    expect(buttonComponent).toMatchSnapshot();
  });
});
