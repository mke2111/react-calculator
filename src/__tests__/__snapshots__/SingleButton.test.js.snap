// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component with snapshot Renders button component correctly 1`] = `
<button
  onClick={[Function]}
  type="button"
>
  
</button>
`;
